packaging/pep517_backend/_manpage_generation.py

```diff
@@ -0,0 +1,112 @@
+"""Manpages generation helpers."""
+
+import os
+import typing as t
+import sys
+from configparser import ConfigParser
+from functools import lru_cache, partial
+from importlib import import_module
+from pathlib import Path
+
+from docutils.core import publish_string
+from jinja2 import Template
+
+
+PROJECT_DIR_PATH = Path(__file__).parents[2]
+
+
+_convert_rst_to_manpage_text = partial(publish_string, writer_name='manpage')
+
+
+@lru_cache(maxsize=1)
+def _make_in_tree_ansible_importable() -> None:
+    """Add the library directory to module lookup paths."""
+    lib_path = str(PROJECT_DIR_PATH / 'lib/')
+    sys.path.insert(0, lib_path)  # NOTE: for the current runtime session
+
+
+@lru_cache(maxsize=1)
+def _make_in_tree_sphinx_extension_importable() -> None:
+    """Add the Sphinx extension directory to module lookup paths."""
+    sphinx_ext_path = str(PROJECT_DIR_PATH / 'docs' / 'docsite' / '_ext')
```

Comment:
Can this be refactored to avoid having a dependency on the `docs` directory in the build backend?

Reply:
@mattclay I wanted to keep the changes small. So yea, there's some dependency on the docs bits that cannot be avoided right now. I thought I'd avoid having copy-pasted duplicate files for now.

Reply:
I'm less concerned about changes to existing files in the `docs` directory. Almost half of the lines added by this PR are new files there. The Sphinx extension and CLI stubs using it should be somewhere under the `packaging` directory. The docs build process should be able to use them without having to duplicate files in the repository.

This will allow us to remove dependencies on the `docs` directory instead of adding more.

Reply:
The split of the `docs` directory into https://github.com/ansible/ansible-documentation has started. That directory is effectively frozen in this repository until its removal. The man page builds for releases and docs are now independent of each other, so this PR can focus entirely on changes to the build backend.